lightning/src/ln/channelmanager.rs

```diff
@@ -1186,7 +1186,9 @@ impl<ChanSigner: ChannelKeys, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref>
 							res.extend_from_slice(&byte_utils::be32_to_array(msg.cltv_expiry));
 						}
 						else if code == 0x1000 | 20 {
-							res.extend_from_slice(&byte_utils::be16_to_array(chan_update.contents.flags));
+							let mut flags = chan_update.contents.flags as u16;
+							if let OptionalField::Present(_) = chan_update.contents.htlc_maximum_msat {flags |= 1 << 8};
```

Comment:
flags is a u8, I think this diff hunk can be droped.

Reply:
`chan_update.contents.flags` is u8, so I get this compile error: `expected `u16`, found `u8``
If I switch to `res.push(upd.contents.flags);`, 4 tests fail.

But also, I don't understand why you're suggesting to omit the `message_flags` part of flags?

Reply:
Oh, interesting, I think this is a misreading of the spec (in the original code). See https://github.com/lightningnetwork/lightning-rfc/issues/791 but my read is that we should likely be pushing two zero bytes, not the flags bytes from the update.

Reply:
Addressed.